Docs: setup_experience_platforms for cross-platform GitOps selection (#49091)
fleetdm/fleet: docs/Configuration/yaml-files.md

        - "🌎 Browsers"
      self_service: true
      setup_experience: true
    - path: ../lib/onboarding-script.sh.package.yml

>       setup_experience_platform: darwin

Reviewer: ```suggestion
      setup_experience_platform: darwin, linux
```

        - "🌎 Browsers"
      self_service: true
      setup_experience: true
    - path: ../lib/onboarding-script.sh.package.yml
      setup_experience_platforms:

>         - macos

Reviewer 2: ```suggestion
    - path: ../lib/onboarding-script.sh.package.yml
      setup_experience_platform: darwin
```

@marko-lisica @cdcme I think we want to use singular "platform" because that's the expected scenario in the real world. Script-only packages are either going to be macOS or Linux. Singular "platform" is also consistent with labels (see below).

Also, instead of a list (array), I think we want to use a command separated list to be consistent with labels.

And, for consistency, I think we use `darwin` instead of `macos`.

<img width="659" height="301" alt="Image" src="https://github.com/user-attachments/assets/a124a9b3-4ea7-4ccf-a16e-fa518959c1d6" />

Author: @noahtalerman / @marko-lisica Latest commit makes these changes to the docs (singular `setup_experience_platform`, comma-separated string, and `darwin` throughout), matching the merged code (#49245) and the query/policy platform convention.

  - Category names support emojis and can be up to 255 characters long. The uniqueness checks ignore emojis, so `"🌎 Browsers"` and `"🔍 Browsers"` are treated as the same name.
  - For Fleet-maintained apps, if `categories` is omitted, apps get their [default categories](https://github.com/fleetdm/fleet/tree/main/ee/maintained-apps/outputs). If `categories` is empty, default categories are removed. If custom categories are specified, apps don't get their default categories unless they're specified explicitly. 
- `setup_experience` installs the software when hosts enroll (default: `false`). On Windows and Linux hosts, if the software has associated policies, Fleet checks them first and skips the install when the host passes all of them. Learn more in the [setup experience guide](https://fleetdm.com/guides/setup-experience).

> - `setup_experience_platform` is a comma-separated string listing every platform whose setup experience should include this package (consistent with the `platform` field on queries and policies). When set, it's the single source of truth for the package's setup experience state and overrides `setup_experience` on the same package. Only `.sh` script-only packages are cross-platform-capable — `.sh` files run on macOS in addition to their native Linux — so the supported values are `darwin` and `linux` (use `darwin`, not `macos`). Examples: `darwin` selects only for macOS (native Linux install-during-setup off); `darwin,linux` selects for both; `linux` is equivalent to `setup_experience: true`; an empty string clears any prior selection. Omit the field entirely to leave existing selections unchanged.

Reviewer: ```suggestion
- `setup_experience_platform` specifies which platform to target for the `.sh` script-only packages in setup experience. Choices for `platform` are `darwin` and `linux`. If not specified and `setup_experience` is `true`, Linux is the default platform.
```

Reviewer: @cdcme I simplified this description. I'm wondering what is a default behavior today, if you specify `setup_experience: true`. Does it only make that package available for the Linux setup experience?

If that's true, then I think we should change "All platforms are targeted if not specified" to "Linux is targeted if not specified", so we don't break this for users who use this feature already.

Author: @marko-lisica Yes, `setup_experience: true` on a `.sh` targets Linux only (its native platform); macOS comes solely from `setup_experience_platform: darwin`. So "all platforms" is wrong. Maybe "If omitted, `setup_experience` governs membership, targeting the native platform (Linux for a `.sh`)." (Plain "Linux if not specified" only holds when `setup_experience: true`; otherwise nothing is targeted.)

Reviewer: @cdcme I updated this. Let me know if this reflects what you built.

Author: @marko-lisica Exactly, yes!

  - Category names support emojis and can be up to 255 characters long. The uniqueness checks ignore emojis, so `"🌎 Browsers"` and `"🔍 Browsers"` are treated as the same name.
  - For Fleet-maintained apps, if `categories` is omitted, apps get their [default categories](https://github.com/fleetdm/fleet/tree/main/ee/maintained-apps/outputs). If `categories` is empty, default categories are removed. If custom categories are specified, apps don't get their default categories unless they're specified explicitly. 
- `setup_experience` installs the software when hosts enroll (default: `false`). On Windows and Linux hosts, if the software has associated policies, Fleet checks them first and skips the install when the host passes all of them. Learn more in the [setup experience guide](https://fleetdm.com/guides/setup-experience).

> - `setup_experience_platforms` lists every platform whose setup experience should include this package. When set, it's the single source of truth for the package's setup experience state and overrides `setup_experience` on the same package. Currently only `.sh` script-only packages are cross-platform-capable — `.sh` files run on macOS via `/bin/sh` in addition to their native Linux — so the supported list values are `macos` and `linux`. Examples: `[macos]` selects only for macOS (native Linux install-during-setup off); `[macos, linux]` selects for both; `[linux]` is equivalent to `setup_experience: true`; `[]` clears any prior selection. Omit the field entirely to leave existing selections unchanged.

Reviewer 3: re: Noah's comment above: 
```suggestion
- `setup_experience_platforms` lists every platform whose setup experience should include this package. When set, it's the single source of truth for the package's setup experience state and overrides `setup_experience` on the same package. Currently only `.sh` script-only packages are cross-platform-capable — `.sh` files run on macOS via `/bin/sh` in addition to their native Linux — so the supported list values are `macos` and `linux`. Examples: `"darwin"` selects only for macOS (native Linux install-during-setup off); `"darwin,linux"` selects for both; `"linux"` is equivalent to `setup_experience: true`; `""` clears any prior selection. Omit the field entirely to leave existing selections unchanged.
```

Author: @rachaelshaw Sorry, I pushed some changes before seeing these comments! Let me know if you want changes to the most recent working, happy to do that.
